src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public class Lottery {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Ball getRandomBall() {
+        return new Ball(colorSupplier.getRandomColor(), colorSupplier.getRandom().nextInt(101));
```

Comment:
You are creating a new instance of Random each time you call getRandomColor(). Consider creating a single instance of Random at the class level to avoid unnecessary object creation.